synapse_invite_checker/invite_checker.py

```diff
@@ -906,13 +938,32 @@ async def get_timestamp_of_last_eligible_activity_in_room(
             limit=5,
             event_filter=event_filter,
         )
-        # If we succeeded with event type filtering, all of these should be only messages
-        last_timestamp = None
-        collected_timestamps = {event_base.origin_server_ts for event_base in events}
-        if collected_timestamps:
-            last_timestamp = max(collected_timestamps)
 
-        return last_timestamp
+        return events
+
+    async def last_invite_ts_in_room(self, room_id: str) -> int | None:
+        """
+        Retrieve the newest origin_server_ts for a membership event in a given room when
+        the sender key does not match the state_key. This is going to be most likely
+        an invite
+        """
+        # Unfortunately, there does not appear to be a nice helper to select a row when
+        # the condition needs to have a `!=` compare clause
+        sql = """
+        SELECT origin_server_ts
+        FROM events
```

Comment:
```suggestion
        sql = """
        SELECT MAX(origin_server_ts)
        FROM events
```

Wouldn't that be way more efficient? :D